src/main/java/com/epam/java/training/controller/ProductController.java

```diff
@@ -0,0 +1,286 @@
+			@ApiResponse(code=401, message="Bad Request"),
+			@ApiResponse(code=402, message="Unauthorised"),
+			@ApiResponse(code=403, message="Forbidden"),
+			@ApiResponse(code=404, message="No Product is found"),
+			@ApiResponse(code=500, message="Internal Server Error")
+	})
+	public ResponseEntity<Product> getProductById(@PathVariable("id") Integer id) {
+		logger.info("Before Review response received====....");
+		Product product = productService.getProductById(id);
+
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_JSON);
+		RestTemplate restTemplate = new RestTemplate();
+		headers.set("API_KEY", "12345");
+		String url = "http://localhost:8080/{productId}/reviews";
+		HttpEntity<String> requestEntity = new HttpEntity<String>(headers);
+		ResponseEntity<Review[]> responseEntity = null;
+		try {
+			responseEntity = restTemplate.exchange(url, HttpMethod.GET, requestEntity, Review[].class, id);
+		} catch (RestClientException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		List<Review> productReviews = new ArrayList<Review>();
+		if (responseEntity != null) {
+			Review[] reviews = responseEntity.getBody();
+			logger.info("Review response received...." + reviews);
+
+			for (Review review : reviews) {
+				logger.info("Review Description:" + review.getDescription() + ", Rating:" + review.getRating());
+				productReviews.add(review);
+			}
+		}
+		if (product != null) {
+			product.setReviews(productReviews);
+		}
+		return new ResponseEntity<Product>(product, HttpStatus.OK);
+	}
+
+	@GetMapping("products")
+	@ApiOperation(value="Products Information", notes="Provides all Products Information.", response=ResponseEntity.class)
+	@ApiResponses(value={
+			@ApiResponse(code=200, message="Success"),
+			@ApiResponse(code=401, message="Bad Request"),
+			@ApiResponse(code=402, message="Unauthorised"),
+			@ApiResponse(code=403, message="Forbidden"),
+			@ApiResponse(code=404, message="No Product is found"),
+			@ApiResponse(code=500, message="Internal Server Error")
+	})
+	public ResponseEntity<List<Product>> getAllProducts() {
+		List<Product> list = productService.getAllProducts();
+
+		List<Product> allProductReviews = new ArrayList<Product>();
+		Product product = null;
+		if (list != null && list.size() > 0) {
+			Iterator<Product> iteratorProduct = list.iterator();
+			while (iteratorProduct.hasNext()) {
+				product = (Product) iteratorProduct.next();
+
+				HttpHeaders headers = new HttpHeaders();
+				headers.setContentType(MediaType.APPLICATION_JSON);
+				RestTemplate restTemplate = new RestTemplate();
+				headers.set("API_KEY", "12345");
+				String url = "http://localhost:8080/{productId}/reviews";
+				HttpEntity<String> requestEntity = new HttpEntity<String>(headers);
+				ResponseEntity<Review[]> responseEntity = null;
+				try {
+					responseEntity = restTemplate.exchange(url, HttpMethod.GET, requestEntity, Review[].class,
```

Comment:
Please move this heavy lifting logic to service layer

Reply:
Completed